src/org/traccar/database/CommandsManager.java

```diff
@@ -31,26 +35,22 @@
 
 public class CommandsManager  extends ExtendedObjectManager<Command> {
 
-    private boolean fallbackToText;
+    private final Map<Long, Queue<Command>> deviceQueues = new ConcurrentHashMap<>();
```

Comment:
I know queue sounds like a right choice here, but really we can as well use regular list here. As soon as device is online we send all commands anyway.